ChatApp/Controlers/ChatViewController.swift

```diff
@@ -0,0 +1,125 @@
+//  ChatApp
+//
+//  Created by bakhva  on 19.04.23.
+//
+
+import UIKit
+
+final class ChatViewController: UIViewController {
+    
+    // MARK: Outlets
+    private let dividerView: UIView = {
+        let view = UIView()
+        view.backgroundColor = .yellowViewColor
+        view.layer.masksToBounds = true
+        view.translatesAutoresizingMaskIntoConstraints = false
+       
+        return view
+    }()
+        
+    private let switcherButton = SwitcherButton()
+    
+    private let firstChatView = ChatView()
+    
+    private let secondChatView = ChatView()
+    
+    // MARK: ViewDidLoad
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        view.backgroundColor = .systemBackground
+        setUpUI()
+        addSubViews()
+        setUpConstraints()
+    }
+    
+    //MARK: Setting Up Constratins
+    private func setUpConstraints() {
+        NSLayoutConstraint.activate([
+            // MARK: Screen Split Constraints
+            dividerView.centerXAnchor.constraint(equalTo: view.centerXAnchor),
+            dividerView.centerYAnchor.constraint(equalTo: view.centerYAnchor),
+            dividerView.widthAnchor.constraint(equalTo: view.widthAnchor),
+            dividerView.heightAnchor.constraint(equalToConstant: ChatViewControllerConstants.dividerViewHeight),
+            
+            // MARK: Switch Constraints
+            switcherButton.trailingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.trailingAnchor,
+                                                 constant: -(view.frame.width * 0.032)),
+            switcherButton.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor,
+                                                constant: ChatViewControllerConstants.switcherTopPadding),
+            switcherButton.widthAnchor.constraint(equalToConstant: ChatViewControllerConstants.switcherWidth),
+            switcherButton.heightAnchor.constraint(equalToConstant: ChatViewControllerConstants.switcherHeight),
+            
+            // MARK: First Chat View Constraints
+            firstChatView.topAnchor.constraint(equalTo: switcherButton.bottomAnchor),
+            firstChatView.trailingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.trailingAnchor),
+            firstChatView.leadingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leadingAnchor),
+            firstChatView.bottomAnchor.constraint(equalTo: dividerView.topAnchor, constant: ChatViewControllerConstants.firstChatViewBottomPadding),
+            
+            //MARK: Second Chat View Constraints
+            secondChatView.topAnchor.constraint(equalTo: dividerView.bottomAnchor),
+            secondChatView.trailingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.trailingAnchor),
+            secondChatView.leadingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leadingAnchor),
+            secondChatView.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor)
+            
```

Comment:
setUpConstraints is very long functions, for readability purposes personally I prefer to decompose: setUpDividerViewConstraints(), setUpSwitcherButtonConstraints(), setUpFirstCharView(), setUpSecondChatView(), besides that marks would not be needed in the scope